General site updates

lucperkins opened the pull request:
This PR makes some a handful of changes to the current site:

* `README` update with some new instructions
* Sets a specific Jekyll version in the `Gemfile`
* Uses a vendored approached to gem installation (to avoid problems stemming from global gems)
* Adds syntax highlighting support for Markdown files
* Fixes the CSS of the `/docs` landing page to force a sticky footer
* Makes the currently available versions of the site data drive (`_data/versions.yml`)
* Makes the copyright date in the footer template driven (it currently reads 2017)

cc @mattklein123

mattklein123 commented:
Thanks @lucperkins! @richarddli @sipplified do you mind taking a look?

richarddli commented on index.html:
```diff
@@ -19,7 +19,7 @@ <h1><img src="{{ hero.image }}" alt="Envoy"></h1>
       <div class="action-btns text-center">
         <a href="https://www.envoyproxy.io/docs/envoy/latest/start/start" class="btn btn-primary">Get Started</a>
-        <a href="https://www.envoyproxy.io/docs/envoy/latest/install/building" class="btn btn-secondary">Download</a>
+        <a href="/download" class="btn btn-secondary">Download</a>
```
We had switched to this link because my theory was that it's more maintained / tested, and I didn't want to maintain a separate Envoy quick start in the web site.

richarddli commented on index.html:
```diff
@@ -111,7 +111,7 @@ <h1 class="heading-big text-uppercase text-purple text-center">Features</h1>
         <a href="https://envoyproxy.github.io/envoy/index.html" class="btn btn-primary">Get Started</a>
-        <a href="download" class="btn btn-secondary">Download</a>
+        <a href="/download" class="btn btn-secondary">Download</a>
       </div>
```
This is a good catch. I think we should point to the official docs for now though, but open to discussion.

lucperkins commented:
@richarddli I think that I would like to improve the "Getting started" story in the official Sphinx docs in general, but I'm happy to address that separately. For now I've removed the `download.md` to prevent future confusion and linked to the official docs. I've also removed the syntax highlighting resources that I added, as it sounds like those won't be necessary without the `download.md` page.
